src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -34,8 +34,9 @@ public static class TarUtils
             {
                 File.Delete(archiveFile);
             }
+
```

Comment:
To avoid scattering checks all over and to make it easier to add other platform/tool-specific "workarounds" in the future, I would consider isolating all of this logic for deciding what process to launch into a single function. Same for untarring Something like this:

```cs
public static ProcessStartInfo GetCreateTarProcessInfo(string archiveFileName, string inputPath)
{
}
```

```cs
public static ProcessStartInfo GetExtractTarProcessInfo(string archiveFileName, string targetPath)
{
}
```

Reply:
Refactored the code.